lib/faraday/options/proxy_options.rb

```diff
@@ -19,6 +19,13 @@ def self.from(value)
           value[:uri] = Utils.URI(uri)
         end
       end
+
+      # URIs without a scheme should default to http (like 'example:123'). This
+      # fixes #1282 and prevents a silent failure in some adapters.
+      if value && !value[:uri].to_s.include?('://')
+        value[:uri] = Utils.URI("http://#{value[:uri]}")
+      end
```

Comment:
What do you think of having this check only in the `case when String` branch?
That looks like the most common use case, and if a `URI` or `Options` object is provided then probably the caller knows what they're doing (leaving the door open to override this automatic fix, if necessary)

Reply:
Good idea - how about this?